packages/terra-form-radio/tests/jest/RadioField.test.jsx

```diff
@@ -8,12 +8,29 @@ import Radio from '../../src/Radio';
 
 window.matchMedia = () => ({ matches: true });
 
+let userAgentGetter;
+beforeEach(() => {
+  userAgentGetter = jest.spyOn(window.navigator, 'userAgent', 'get');
+});
+
 it('should render a default radio field', () => {
   const radioField = (<RadioField legend="Default RadioField" />);
   const wrapper = shallowWithIntl(radioField);
   expect(wrapper).toMatchSnapshot();
 });
 
+it('should render radio field with div element for safari browser or Edg browser', () => {
+  userAgentGetter.mockReturnValue('safari Edg');
+  const wrapper = shallowWithIntl(<RadioField legend="Coustom Message RadioField" />);
+  expect(wrapper.dive()).toMatchSnapshot();
+});
+
+it('should render radio field with legend element for Chrome browser', () => {
+  userAgentGetter.mockReturnValue('Chrome');
+  const wrapper = shallowWithIntl(<RadioField legend="Coustom Message RadioField" />);
+  expect(wrapper.dive()).toMatchSnapshot();
```

Comment:
As @sdadn  mentioned on Checkbox Field PR we can have expect() to ensure component is rendered with `legend` element. Probably we can also use [toMatchInlineSnapshots()](https://jestjs.io/docs/snapshot-testing#inline-snapshots) which ensures component is rendered with expected dom elements

Reply:
I am using .dive() method, it will executes as we expected dom elements.like div or legend based on mockreturn value.